File: src/Woong.MonitorStack.Server/Data/FocusSessionEntity.cs
namespace Woong.MonitorStack.Server.Data;

public sealed class FocusSessionEntity
{
    public long Id { get; set; }

    public Guid DeviceId { get; set; }

    public string ClientSessionId { get; set; } = "";

    public string PlatformAppKey { get; set; } = "";

    public DateTimeOffset StartedAtUtc { get; set; }

    public DateTimeOffset EndedAtUtc { get; set; }

    public long DurationMs { get; set; }

    public DateOnly LocalDate { get; set; }

    public string TimezoneId { get; set; } = "";

    public bool IsIdle { get; set; }

    public string Source { get; set; } = "";

    public int? ProcessId { get; set; }

    public string? ProcessName { get; set; }

    public string? ProcessPath { get; set; }

    public long? WindowHandle { get; set; }

    public string? WindowTitle { get; set; }
}

File: android/app/src/main/java/com/woong/monitorstack/data/local/FocusSessionEntity.kt
package com.woong.monitorstack.data.local

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "focus_sessions")
data class FocusSessionEntity(
    @PrimaryKey val clientSessionId: String,
    val packageName: String,
    val startedAtUtcMillis: Long,
    val endedAtUtcMillis: Long,
    val durationMs: Long,
    val localDate: String,
    val timezoneId: String,
    val isIdle: Boolean,
    val source: String
)
